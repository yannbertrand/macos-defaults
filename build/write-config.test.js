const fs = require('fs')
jest.mock('fs')

const writeConfig = require('./write-config')
require('./handlebars-helpers')

const templatesPath = 'templates'
const destinationPath = 'dist'
const copiedFiles = [
  '.vuepress/public/android-chrome-192x192.png',
  '.vuepress/public/android-chrome-512x512.png',
  '.vuepress/public/apple-touch-icon-120x120-precomposed.png',
  '.vuepress/public/apple-touch-icon-120x120.png',
  '.vuepress/public/apple-touch-icon-152x152-precomposed.png',
  '.vuepress/public/apple-touch-icon-152x152.png',
  '.vuepress/public/apple-touch-icon-precomposed.png',
  '.vuepress/public/apple-touch-icon.png',
  '.vuepress/public/browserconfig.xml',
  '.vuepress/public/favicon.ico',
  '.vuepress/public/favicon-16x16.png',
  '.vuepress/public/favicon-32x32.png',
  '.vuepress/public/logo.svg',
  '.vuepress/public/media-1x1.jpg',
  '.vuepress/public/media-1x1.webp',
  '.vuepress/public/media-2x1.jpg',
  '.vuepress/public/media-2x1.webp',
  '.vuepress/public/mstile-150x150.png',
  '.vuepress/public/netlify.svg',
  '.vuepress/public/robots.txt',
  '.vuepress/public/safari-pinned-tab.svg',
  '.vuepress/public/site.webmanifest',
  '.vuepress/styles/index.styl',
  '.vuepress/styles/palette.styl',
]

describe('write-config', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  beforeEach(() => {
    callWriteConfig()
  })

  it('should copy some static files', () => {
    copiedFiles.forEach((file) => {
      const fileContent = readFile(`${destinationPath}/${file}`)
      expect(fileContent).toEqual(`copied:${templatesPath}/${file}`)
    })
  })

  it('should write the netlify _headers file', () => {
    const netlifyHeadersContent = readFile(
      `${destinationPath}/.vuepress/public/_headers`
    )
    expect(netlifyHeadersContent).toMatchSnapshot()
  })

  it('should write a vuepress config.yml file using the template', () => {
    const vuepressConfigContent = readFile(
      `${destinationPath}/.vuepress/config.yml`
    )
    expect(vuepressConfigContent).toMatchSnapshot()
  })
})

const supportedLanguages = {
  languages: [
    {
      url: '/',
      lang: 'en-US',
      home: 'Home',
      defaults: {
        categories: [
          {
            folder: 'category1',
            name: 'Category 1',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
          {
            folder: 'category2',
            name: 'Category 2',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
        ],
      },
    },
    {
      url: '/fr/',
      lang: 'fr-FR',
      home: 'Accueil',
      defaults: {
        categories: [
          {
            folder: 'categorie1',
            name: 'Catégorie 1',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
          {
            folder: 'categorie2',
            name: 'Catégorie 2',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
        ],
      },
    },
  ],
}

const algoliaConf = {
  ALGOLIA_API_KEY: 'SAMPLE_ALGOLIA_API_KEY',
  ALGOLIA_INDEX_NAME: 'SAMPLE_ALGOLIA_INDEX_NAME',
  ALGOLIA_APP_ID: 'SAMPLE_ALGOLIA_APP_ID',
}

const config = { ...supportedLanguages, ...algoliaConf }

const callWriteConfig = () =>
  writeConfig(config, templatesPath, destinationPath)
const readFile = (file) => fs.readFakeFileSync(file, 'utf8')
