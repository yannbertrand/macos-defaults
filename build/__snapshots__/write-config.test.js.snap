// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-config should write a vuepress config.yml file using the template 1`] = `
"---

title: 'macOS defaults'
description: 'Uncomplete list of macOS defaults commands with demos ✨'
head:
  -
    - link
    - rel: "preload"
      href: "/logo.svg"
      as: "image"
  -
    - link
    - rel: "apple-touch-icon"
      sizes: "180x180"
      href: "/apple-touch-icon.png"
  -
    - link
    - rel: "icon"
      type: "image/png"
      sizes: "32x32"
      href: "/favicon-32x32.png"
  -
    - link
    - rel: "icon"
      type: "image/png"
      sizes: "16x16"
      href: "/favicon-16x16.png"
  -
    - link
    - rel: "manifest"
      href: "/site.webmanifest"
  -
    - link
    - rel: "mask-icon"
      href: "/safari-pinned-tab.svg"
      color: "#004bbd"
  -
    - meta
    - name: 'viewport'
      content: 'width=device-width,initial-scale=1'
  -
    - meta
    - name: "msapplication-TileColor"
      content: "#da532c"
  -
    - meta
    - name: "theme-color"
      content: "#ffffff"
  -
    - meta
    - http-equiv: "Content-Type"
      content: "text/html; charset=utf-8"
  -
    - meta
    - property: "twitter:site"
      content: "@macos_defaults"
  -
    - meta
    - property: "twitter:creator"
      content: "@_YannBertrand"
  -
    - meta
    - property: "og:site_name"
      content: "macOS defaults"
  -
    - meta
    - property: "og:url"
      content: "https://macos-defaults.netlify.app"
  -
    - meta
    - property: "og:type"
      content: "website"
  -
    - meta
    - property: "og:image:alt"
      content: ""
  -
    - meta
    - name: "author"
      content: "Yann Bertrand"
evergreen: true
plugins:
  sitemap:
    hostname: 'https://macos-defaults.com'
    changefreq: 'weekly'
    exclude: ['/404.html']
locales:
  /:
    lang: 'en-US'
  /fr/:
    lang: 'fr-FR'
themeConfig:
  repo: 'yannbertrand/macos-defaults'
  logo: '/logo.svg'
  algolia:
    apiKey: 'SAMPLE_ALGOLIA_API_KEY'
    indexName: 'SAMPLE_ALGOLIA_INDEX_NAME'
    appId: 'SAMPLE_ALGOLIA_APP_ID'
  locales:
    /:
      nav:
        - text: 'Home'
          link: '/'
      activeHeaderLinks: false
      sidebar:
        - title: 'Home'
          path: '/'
        - title: Category 1
          collapsable: false
          path: '/category1/'
          children:
            - '/category1/page'
        - title: Category 2
          collapsable: false
          path: '/category2/'
          children:
            - '/category2/page'
    /fr/:
      nav:
        - text: 'Accueil'
          link: '/fr/'
      activeHeaderLinks: false
      sidebar:
        - title: 'Accueil'
          path: '/fr/'
        - title: Catégorie 1
          collapsable: false
          path: '/fr/categorie1/'
          children:
            - '/fr/categorie1/page'
        - title: Catégorie 2
          collapsable: false
          path: '/fr/categorie2/'
          children:
            - '/fr/categorie2/page'
"
`;

exports[`write-config should write the netlify _headers file 1`] = `
"/*
  Content-Security-Policy: default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'sha256-47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=' 'sha256-PjBkwE8xcYZAp+HsnzzOVNqa/Ra+/v1Fnx6f0PW6ic4=' 'unsafe-hashes' 'sha256-aqNNdDLnnrDOnTNdkJpYlAxKVJtLt9CtFLklmInuUAE=' 'sha256-OFZH/KmE1spbRSh4CjNax9vyG1iEwWYtQyOeXApqoN4=' 'sha256-laxniBKFE69CPUpLCGftSNNcrHa/q3ajB6sagR6TudM='; img-src 'self' data: https://www.netlify.com; media-src 'self' data:; connect-src https://*.algolia.net https://*.algolianet.com;
  X-Frame-Options: DENY
  X-Content-Type-Options: nosniff
  X-XSS-Protection: 1; mode=block
"
`;
